lib/fprotect/Kconfig

```diff
@@ -11,12 +11,19 @@ config FPROTECT
 	  Use hardware peripherals (BPROT, ACL, or SPU) to write-protect
 	  regions of flash until next reset.
 
+
+if !ARM
+config NRF_SPU_FLASH_REGION_SIZE
+	hex
+	default 0x1000
+	help
+	  Redefinition for the benefit of qemu_x86
+endif
+
 if FPROTECT
 config FPROTECT_BLOCK_SIZE
-	int "Block size of fprotect"
-	default 32768 if SOC_NRF9160
-	default 16384 if SOC_NRF5340_CPUAPP
-	default $(dt_int_val,DT_SOC_NV_FLASH_1000000_ERASE_BLOCK_SIZE) if SOC_NRF5340_CPUNET
-	default $(dt_int_val,DT_SOC_NV_FLASH_0_ERASE_BLOCK_SIZE)
+	hex "Block size of fprotect"
+	default NRF_SPU_FLASH_REGION_SIZE if SOC_NRF9160 || SOC_NRF5340_CPUAPP
```

Comment:
This is not runtime, do we need this considering the changes in fprotect_spu.c?

Reply:
I see now that this is used in the other commit. Is it not possible to check more specific details than the SOC, to know that this is indeed the variant that has the issue? In the future I guess this code will have trouble when the bug is fixed.

Reply:
The intent is that this will work when the bug is fixed as well. There is a runtime check in fprotect_spu.c

    #define SPU_BLOCK_SIZE (nrf53_has_erratum19() ? 32*1024 : 16*1024)

`FPROTECT_BLOCK_SIZE` will then be a sort of build-time "worst case" config for use e.g. when aligning things.